pkg/lb/serviceLB/serviceLB.go

```diff
@@ -0,0 +1,35 @@
+package serviceLB
+
+import (
+	ctlCorev1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+
+	lbv1 "github.com/harvester/harvester-load-balancer/pkg/apis/loadbalancer.harvesterhci.io/v1alpha1"
+	ctlDiscoveryv1 "github.com/harvester/harvester-load-balancer/pkg/generated/controllers/discovery.k8s.io/v1beta1"
+	"github.com/harvester/harvester-load-balancer/pkg/lb/entry"
+	"github.com/harvester/harvester-load-balancer/pkg/lb/forwarder"
+)
+
+// ServiceLB takes Kubernetes service as the traffic entry and endpointSlice as the forwarder.
+type ServiceLB struct {
+	lb            *lbv1.LoadBalancer
+	service       *entry.Service
+	endpointSlice *forwarder.CustomEndpointSlice
+}
```

Comment:
The data structure is so confusing.
We have the LB type, then the Service type which wraps a client and a LB(and it's called config),
then the CustomEndpointSlice type which wraps a client and a LB,
then the ServiceLB which wraps the LB, Service, and CustomEndpointSlice?

Reply:
It may be better to pass the LB as a parameter.

Reply:
What if we do it in the following way:
```
type Manager struct{}

func (m Manager) EnsureLB(lb LB){
	EnsureService(lb)
	EnsureEndpointSlice(lb)
}

func (m Manager) EnsureService(lb LB){}

func (m Manager) EnsureEndpointSlice(lb LB){}
```

We wrap a lot of confusing types and it seems to be over-engineering to me.